01-MyMath/MyMath.js

```diff
@@ -15,30 +15,88 @@ export class MyMath {
         // 'this' referenziert den Kontext in dem die aktuelle Funktion aufgerufen wird. 
         // Hier referenziert es die Instanz der Klasse MyMath die wir gerade erstellen.
         // mit 'value * 1' erzwingen wir, dass value als number gelesen wird.
-        this.value = value * 1; 
+        if (!value) {
+            this.value = 0;
+        }
+        else {
+            this.value = value * 1; 
+        }
     }
 
-    add(value) {
+    //Gives an Alert with actual value
+    show() {
+        console.log("Value is " + this.value);
+    }
 
+    get() {
+        return this.value;
     }
 
-    subtract(value) {
+    // Set the value of myMath manually:
+    set(value) {
+        this.value = value;
+        return this;
+    }
 
+    add(value) {
+        this.value += value;
+        return this;
     }
 
-    multiply(value) {
+    subtract(value) {
+        this.value -= value;
+        return this;
+    }
 
+    multiply(value) {
+        this.value *= value;
+        return this;
     }
 
     divide(value) {
+        if (value == 0) {
+            if (confirm("You are about to divide by ZERO, do you know what you`re doing?!?")) {
+                alert("Action canceled to prevent cracks in the space-time continuum!");
+              } else {
 
+              }
+        }
+        else {
+            this.value /= value;
+        }
+
+        return this;
     }
 
     pow(value) {
+        let temp = this.value;
+
+        if (value == 0) {
```

Comment:
```suggestion
        if (value === 0) {
```
wir wollen ja nicht `[]`,`""`,`"0"` oder `false` matchen

Reply:
btw: `a = { value: 0 }; a.valueOf = () => a.value; a == 0 (true) `